### PR TITLE
Preare Update Java JDK from 21 to 22

### PR DESCRIPTION
**Pull request by koppor**
Refs https://github.com/JabRef/jabref-issue-melting-pot/issues/669

image

### Mandatory checks

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if change is visible to the user)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: .github/workflows/deployment-jdk-ea.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest, buildjet-8vcpu-ubuntu-2204-arm]
-        jdk: [22]
+        jdk: [23]
```
Comment (Siedlerchr) [inline .github/workflows/deployment-jdk-ea.yml:40]: shouldn't EA become 24 then?

**Review by Siedlerchr — DISMISSED** [review]

**Comment by Siedlerchr** [comment]
> Unsupported class file major version 67

File: .github/workflows/deployment-jdk-ea.yml (hunk 1 above)
Comment (koppor) [reply .github/workflows/deployment-jdk-ea.yml:40]: Not supported by the current workflow. I can re-add this, but takes 30 to 60 min again - need to switch to Oracle action etc.

**Comment by koppor** [comment]
Blocked by https://github.com/beryx/badass-jlink-plugin/issues/275.

I don't want to use a patched version of the `jlink` plugin. Thus, I close this PR.

**Comment by koppor** [comment]
Since I also improved the links for updating and we should try if all dependencies work with a JDK "close enough" to JDK23, I "downgraded" to JDK 22 in this "upgrade"

**Comment by Siedlerchr** [comment]
JDK 22 does not make sense, it's obsolete with the release of 23.

**Comment by koppor** [comment]
> JDK 22 does not make sense, it's obsolete with the release of 23.

Ah, yeah, I totally forgot.

Now, I just have the documentation updates in - and some small version adaptions.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -309,7 +309,7 @@ jobs:
         if: github.ref == 'refs/heads/main'
         uses: actions/setup-java@v4
         with:
-          java-version: 22.0.2
+          java-version: 21.0.2
```
Comment (calixtus) [inline .github/workflows/tests.yml]: 21.0.5 ?

**Review by calixtus — APPROVED** [review]

**Pull request merged**